improc/simulator.py

```diff
@@ -802,29 +890,582 @@ def make_galaxy_image(
         rotation = np.deg2rad(rotation)
 
         # regular coordinate grid
+        t0 = time.time()
         x0, y0 = np.meshgrid(np.arange(imsize[1]), np.arange(imsize[0]))
+        cls.runtime['meshgrid'] += time.time() - t0
 
         # transform the coordinates using rotation and translation
+        t0 = time.time()
         x = (x0 - center_x) * np.cos(rotation) + (y0 - center_y) * np.sin(rotation)
         y = (y0 - center_y) * np.cos(rotation) - (x0 - center_x) * np.sin(rotation)
         r = np.sqrt(x ** 2 + y ** 2)
+        cls.runtime['transform'] += time.time() - t0
 
         # first make the bulge using a sersic profile
+        t0 = time.time()
         r0 = sersic_scale
-        bulge = np.exp(-7.67 * (r / r0) ** (1 / 4))
+        bulge = np.exp( -7.67 * np.power(r / r0, 1 / 4) )
         bulge *= sersic_flux / np.sum(bulge)
+        cls.runtime['bulge'] += time.time() - t0
 
         # now make the disk
+        t0 = time.time()
         r0 = exp_scale
-        disk = np.exp(-1.67 * np.sqrt((x / r0 / cos_i) ** 2 + (y / r0) ** 2))
+        disk = np.exp(-1.67 * np.sqrt( (x / r0 / cos_i) ** 2 + (y / r0) ** 2) )
         disk *= exp_flux / np.sum(disk)
-        disk *= cos_i  # correct for the projection of the disk
+        cls.runtime['disk'] += time.time() - t0
 
         # add them together
+        t0 = time.time()
         galaxy_image = bulge + disk
+        cls.runtime['add'] += time.time() - t0
+
+        # add cutoff
+        if cutoff_radius is not None:
+            t0 = time.time()
+            cutoff = np.ones(galaxy_image.shape)  # inside the radius there is no change
+            # use the disk scale as the length scale for the cutoff
+            cutoff[r > cutoff_radius] = np.exp(-5.0 * (r[r > cutoff_radius] - cutoff_radius) / exp_scale)
+            galaxy_image *= cutoff
+            cls.runtime['cutoff'] += time.time() - t0
```

Comment:
This is fine, no need to change it, but I use time.perf_counter() for this sort of thing -- that's what the time documentation suggests is best for this kind of thing.  I think unless you really want high resolution, it doesn't matter that much.

Reply:
Yes I will start using that.